### PR TITLE
Update list scrolling behaviour

### PR DESCRIPTION
**Pull request by stuartleeks**
When scrolling up and down long lists the behaviour seemed a little unintuitive. This change hopefully makes the behaviour feel more natural

### CODE REVIEW

**Comment by lawrencegripper** [comment]
When opening the `deployments` response on a particular RG I see the following crash. Just trying to work out what the root cause is. 

```
22:07 $ azbrowse
AzBrowse is waiting for access to '~/.azbrowse.db', do you have another instance of azbrowse open?
Loading db complete
2019/08/18 22:07:34 Current version is the latest
A crash occurred: runtime error: slice bounds out of rangegoroutine 1 [running]:
runtime/debug.Stack(0x3a, 0x0, 0x0)
	/snap/go/4289/src/runtime/debug/stack.go:24 +0x9d
runtime/debug.PrintStack()
	/snap/go/4289/src/runtime/debug/stack.go:16 +0x22
main.main.func2()
	/home/lawrence/go/src/github.com/lawrencegripper/azbrowse/cmd/azbrowse/main.go:107 +0xad
panic(0xa67cc0, 0x1329fe0)
	/snap/go/4289/src/runtime/panic.go:522 +0x1b5
github.com/lawrencegripper/azbrowse/internal/pkg/views.(*ListWidget).Layout(0xc0000ac270, 0xc0000d8090, 0x0, 0x0)
	/home/lawrence/go/src/github.com/lawrencegripper/azbrowse/internal/pkg/views/treeview.go:95 +0x7ef
github.com/jroimartin/gocui.(*Gui).flush(0xc0000d8090, 0x0, 0x0)
	/home/lawrence/go/pkg/mod/github.com/jroimartin/gocui@v0.4.0/gui.go:435 +0xcc
github.com/jroimartin/gocui.(*Gui).MainLoop(0xc0000d8090, 0xbb29e8, 0xc00007e2a0)
	/home/lawrence/go/pkg/mod/github.com/jroimartin/gocui@v0.4.0/gui.go:384 +0x227
main.main()
	/home/lawrence/go/src/github.com/lawrencegripper/azbrowse/cmd/azbrowse/main.go:222 +0x15e4
To capture a more detailed train run 'azbrowse --debug' and reproduce the issue. Visit https://github.com/lawrencegripper/azbrowse/issues to raise a bug
```

**Comment by stuartleeks** [comment]
Is that a RG with no deployments and I've missed a bounds check somewhere?

File: internal/pkg/views/treeview.go [hunk 1]
```diff
@@ -71,11 +72,27 @@ func (w *ListWidget) Layout(g *gocui.Gui) error {
...
+		diff := topIndex - w.selected
+		topIndex -= diff
+		bottomIndex -= diff
+	}
+	w.lastTopIndex = topIndex
+	if bottomIndex > len(allItems) {
+		bottomIndex = len(allItems) - 1
+	}
+
+	for _, item := range allItems[topIndex : bottomIndex+1] {
```
Comment (stuartleeks) [inline internal/pkg/views/treeview.go]: I suspect that this line is trying to be too clever and failing if there are no list items

Comment (lawrencegripper) [reply internal/pkg/views/treeview.go]: Yeah that makes sense to me based on the stacktrace. Sorry I didn't get a the repro as was just randomly jumping around in a sub without `--debug` on and then couldn't find the RG I opened :(

Comment (stuartleeks) [reply internal/pkg/views/treeview.go]: I didn't manage to repro (tried creating a new RG and viewing deployments) but I've tweaked this bit of code to hopefully be more resilient

**Pull request merged**